public/app/plugins/datasource/prometheus/components/AnnotationQueryEditor.tsx

```diff
@@ -19,35 +19,47 @@ export function AnnotationQueryEditor(props: Props) {
   // This is because of problematic typing. See AnnotationQueryEditorProps in grafana-data/annotations.ts.
   const annotation = props.annotation!;
   const onAnnotationChange = props.onAnnotationChange!;
+  const query = { expr: annotation.expr, refId: annotation.name, interval: annotation.step };
+
   return (
     <>
-      <PromQueryEditorSelector
-        {...props}
-        query={{ expr: annotation.expr, refId: annotation.name, interval: annotation.step }}
-        onChange={(query) =>
-          onAnnotationChange({
-            ...annotation,
-            expr: query.expr,
-            step: query.interval,
-          })
-        }
-        uiOptions={{
-          modes: {
-            [QueryEditorMode.Explain]: false,
-            [QueryEditorMode.Code]: true,
-            [QueryEditorMode.Builder]: true,
-          },
-          runQueryButton: false,
-          options: {
-            exemplars: false,
-            type: false,
-            format: false,
-            minStep: true,
-            legend: false,
-            resolution: false,
-          },
-        }}
-      />
+      <EditorRows>
+        <PromQueryCodeEditor
+          {...props}
+          query={query}
+          onChange={(query) => {
+            onAnnotationChange({
+              ...annotation,
+              expr: query.expr,
+            });
+          }}
+        />
+        <EditorRow>
+          <EditorField
+            label="Min step"
+            tooltip={
+              <>
+                An additional lower limit for the step parameter of the Prometheus query and for the{' '}
+                <code>$__interval</code> and <code>$__rate_interval</code> variables.
+              </>
+            }
+          >
+            <AutoSizeInput
+              type="text"
+              aria-label="Set lower limit for the step parameter"
+              placeholder={'auto'}
+              minWidth={10}
+              onCommitChange={(ev) => {
+                onAnnotationChange({
+                  ...annotation,
+                  interval: ev.currentTarget.value,
```

Comment:
Can you confirm `step` and `interval` leads to same context here? Previous code, step property was changed and here `interval ` is being changed. ( aria label reads as step parameter )

Reply:
ah good catch yeah this should be mapped to `step` in the annotation